Source/WebCore/css/parser/CSSParser.cpp

```diff
@@ -1495,7 +1497,9 @@ RefPtr<StyleRuleBase> CSSParser::consumeStyleRule(CSSParserTokenRange prelude, C
 void CSSParser::consumeBlockContent(CSSParserTokenRange range, StyleRuleType ruleType, OnlyDeclarations onlyDeclarations, ParsingStyleDeclarationsInRuleList isParsingStyleDeclarationsInRuleList)
 {
     auto nestedRulesAllowed = [&] {
-        return isStyleNestedContext() && onlyDeclarations == OnlyDeclarations::No;
+        if (onlyDeclarations == OnlyDeclarations::Yes)
+            return false;
+        return isStyleNestedContext() || isFunctionNestedContext();
```

Comment:
`@function` body is a [declaration-rule-list](https://drafts.csswg.org/css-syntax-3/#typedef-declaration-rule-list)

It seems the current patch allow qualified rule (~style rule) in the body ?